version: '3.8'

services:
  postgres:
    image: postgres:16-alpine
    container_name: exposicao-arte-dbdatabase_url
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: exposicao_arte
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - app-network

  app:
    build:
      context: .
      dockerfile: dockerfile
    container_name: exposicao-arte-backend
    environment:
      DATABASE_URL: postgresql://postgres:postgres@postgres:5432/exposicao_arte
    ports:
      - "8080:8080"
    depends_on:
      postgres:
        condition: service_healthy
    networks:
      - app-network
    volumes:
      - .:/app

volumes:
  postgres_data:


networks:
  app-network:
    driver: bridge
